docs/narr/urldispatch.rst

```diff
@@ -1100,52 +1097,73 @@ was matched (see :class:`pyramid.interfaces.IRoute` for the API of such a route
 object).
 
 ``info['match']`` is useful when predicates need access to the route match.
-For example:
+Imagine you want to define a route when a part of the URL matches some
+specific values.  You could define three view configurations, each one
+with its own ``match_param`` value (see :ref:`predicate_view_args`), but
+you want to think of it as one route, and associate it with one view.
+See this code:
 
 .. code-block:: python
     :linenos:
 
-    def any_of(segment_name, *allowed):
-        def predicate(info, request):
-            if info['match'][segment_name] in allowed:
-                return True
-        return predicate
+    class AnyOfPredicate:
+        def __init__(self, val):
+            self.segment_name = val[0]
+            self.allowed = tuple(val[0:])
+
+        def text(self):
+            args = (self.segment_name,) + self.allowed
+            return 'any_of = %s' % (args,)
+
+        phash = text
+
+        def __call__(self, info, request):
+            return info['match'][self.segment_name] in self.allowed
+
 
-    num_one_two_or_three = any_of('num', 'one', 'two', 'three')
+    config.add_route_predicate("any_of", AnyOfPredicate)
+    config.add_route('route_to_num', '/{num}', any_of=('num', 'one', 'two', 'three'))
 
-    config.add_route('route_to_num', '/{num}',
-                     custom_predicates=(num_one_two_or_three,))
 
-The above ``any_of`` function generates a predicate which ensures that the
+An instance of the class defined above is a predicate which ensures that the
 match value named ``segment_name`` is in the set of allowable values
-represented by ``allowed``.  We use this ``any_of`` function to generate a
-predicate function named ``num_one_two_or_three``, which ensures that the
-``num`` segment is one of the values ``one``, ``two``, or ``three`` , and use
-the result as a custom predicate by feeding it inside a tuple to the
-``custom_predicates`` argument to
-:meth:`~pyramid.config.Configurator.add_route`.
+represented by ``allowed``.  We register this class as a *predicate
+factory* with the ``any_of`` argument name, then we can use that new
+keyword argument with :meth:`~pyramid.config.Configurator.add_route` to
+have Pyramid instantiate the class (the ``('num', 'one', 'two',
+'three')`` value is passed to the constructor as the argument ``val``)
+and use the resulting instance as a custom predicate to check if
+incoming requests match the route or not.
```

Comment:
This sentence is a bit long, please advise on better wording!

Reply:
Ah, I am not a fan of long lines (harder editor navigation, less nice to read, sometimes makes worse diffs), but I will try to follow the existing convention.  Thanks for the text, I will merge our versions.